plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java

```diff
@@ -134,30 +132,38 @@ protected IStatus run(IProgressMonitor monitor) {
     }
   }
 
-  /** Creates an appengine-web.xml file in the WEB-INF folder if it doesn't exist. */
-  @VisibleForTesting
-  void createConfigFiles(
-      IProject project, IProjectFacetVersion facetVersion, IProgressMonitor monitor)
-      throws CoreException {
+  void createConfigFiles(IProject project, IProjectFacetVersion facetVersion,
+      IProgressMonitor monitor) throws CoreException {
     SubMonitor progress = SubMonitor.convert(monitor, 10);
 
-    IFile appEngineWebXml = WebProjectUtil.findInWebInf(project, new Path(APPENGINE_WEB_XML));
-    if (appEngineWebXml != null && appEngineWebXml.exists()) {
-      return;
-    }
+    createFileInWebInf(project, "logging.properties", Templates.LOGGING_PROPERTIES_TEMPLATE,
+        Collections.emptyMap(), progress.newChild(5));
 
-    // Use the virtual component model decide where to create the appengine-web.xml
-    appEngineWebXml = WebProjectUtil.createFileInWebInf(project, new Path(APPENGINE_WEB_XML),
-        new ByteArrayInputStream(new byte[0]), progress.newChild(2));
-    String configFileLocation = appEngineWebXml.getLocation().toString();
     Map<String, String> parameters = new HashMap<>();
     Object appEngineRuntime = facetVersion.getProperty("appengine.runtime");
     if (appEngineRuntime instanceof String) {
       parameters.put("runtime", (String) appEngineRuntime);
     }
-    Templates.createFileContent(
-        configFileLocation, Templates.APPENGINE_WEB_XML_TEMPLATE, parameters);
+    createFileInWebInf(project, "appengine-web.xml", Templates.APPENGINE_WEB_XML_TEMPLATE,
+        parameters, progress.newChild(5));
```

Comment:
as of Eclipse 4.6 we don't need to use newChild any more. We can start using split. See https://www.eclipse.org/articles/Article-Progress-Monitors/article.html